### PR TITLE
Trivial: Remove duplicate EOF trailing newlines

### PR DESCRIPTION
**Pull request by MaEtUgR**
**Describe problem solved by this pull request**
When merging master into a branch the and experiencing conflicts the merge can be completed after resolving the conflicts using `git commit` but with the currently existing EOF trailing newlines you get a bunch of messages like `path/filename:linenumber: new blank line at EOF` and it doesn't continue.

**Describe your solution**
Since these newlines are useless anyways I suggest to just remove them.

**Test data / coverage**
It's a trivial change and still compiles.

**Additional context**
I'm still following up to create usefull prs out of #12072 and using the oportunity to learn merging correctly since I have to still keep it up to date to take over missing things.

### CODE REVIEW

**Comment by MaEtUgR** [comment]
> I don't think it's required, since we have astyle, and it is the cause of such issues.

But astyle doesn't get applied to CMake, Python, ... files and then we would suddenly have trailing whitespaces everywhere again not?

**Comment by bkueng** [comment]
> But astyle doesn't get applied to CMake, Python, ... files and then we would suddenly have trailing whitespaces everywhere again not?

Yeah that's a good point. But we would have that problem already since not everyone uses the pre-commit hook and it does not run on CI either. So I don't think it would make things worse.
If we want it, we should add it to CI as well.
Either way works for me.

File: Tools/astyle/pre-commit [hunk 1]
```diff
@@ -49,6 +49,7 @@ fi
 # If there are whitespace errors, print the offending file names and fail.
 git diff-index --check --cached $against --
 if [ $? -ne 0 ]; then
+	echo "Pre-commit style error: Whitespace issues"
```
Comment (ndepal) [inline Tools/astyle/pre-commit]: Style error: you're mixing tabs and spaces :smile:

File: Tools/astyle/pre-commit [hunk 2]
```diff
@@ -58,6 +59,7 @@ do
         ./Tools/astyle/files_to_check_code_style.sh $i | xargs -n 1 -P 8 -I % ./Tools/astyle/check_code_style.sh %
         if [ $? -ne 0 ]
         then
+		echo "Pre-commit style error: Bad formatting according to astyle rules"
```
Comment (ndepal) [inline Tools/astyle/pre-commit]: ditto

File: Tools/astyle/pre-commit (hunk 1 above)
Comment (MaEtUgR) [reply Tools/astyle/pre-commit]: Good point 👍 , vscode detected the prevalently used indentation method in the file and everything I did was with tabs. Now I put a commit to change the rest of the file. It's consistent with the other shell scripts I looked at.

**Pull request merged**